From 52c1b6a61fd7f88bef867ed669d58e3d508f38a6 Mon Sep 17 00:00:00 2001
From: Donald McIntyre <newfination@gmail.com>
Date: Thu, 27 Jun 2019 20:23:56 -0600
Subject: [PATCH 1/2] General adjustments and adaptation to ECIP process
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

• corrected some spelling
• adjusted grammar
• added formatting for clarity
• corrected some links
• added some subtitles for clarity
---
 ECIPs/ECIP-1000.mediawiki | 187 +++++++++++++++++++++-----------------
 1 file changed, 106 insertions(+), 81 deletions(-)

diff --git a/ECIPs/ECIP-1000.mediawiki b/ECIPs/ECIP-1000.mediawiki
index 43c667f4..d0bac6f9 100644
--- a/ECIPs/ECIP-1000.mediawiki
+++ b/ECIPs/ECIP-1000.mediawiki
@@ -14,47 +14,59 @@
 
 An Ethereum Classic Improvement Proposal (ECIP) is a design document providing information to the Ethereum Classic community, or describing a new feature for Ethereum Classic or its processes or environment. The ECIP should provide a concise technical specification of the feature and a rationale for the feature.
 
-We intend ECIPs to be the primary mechanisms for proposing new features, for collecting community input on an issue, and for documenting the design decisions that have gone into Ethereum Classic. The ECIP author is responsible for building consensus within the community and documenting dissenting opinions.
+We intend ECIPs to be the primary mechanism for proposing new features, for collecting community input on an issue, and for documenting the design decisions that have gone into Ethereum Classic. The '''ECIP author''' is responsible for building consensus within the community and documenting dissenting opinions.
 
 Because the ECIPs are maintained as text files in a versioned repository, their revision history is the historical record of the feature proposal.
 
 ==Copyright==
 
-This ECIP is licensed Apache-2, originally by Luke Dashjr <luke+ECIP@dashjr.org> under BSD 2-clause license.
+This ECIP is licensed Apache-2, originally by [https://github.com/luke-jr Luke Dashjr] under BSD 2-clause license.
 
 ==ECIP workflow==
 
-The ECIP process begins with a new idea for Ethereum Classic. Each potential ECIP must have a champion -- someone who writes the ECIP using the style and format described below, shepherds the discussions in the appropriate forums, and attempts to build community consensus around the idea. The ECIP champion (a.k.a. Author) should first attempt to ascertain whether the idea is ECIP-able.
-Small enhancements or patches to a particular piece of software often don't require standardisation between multiple projects; these don't need an ECIP and should be injected into the relevant project-specific development workflow with a patch submission to the applicable issue tracker.
+===Introduction===
+
+The ECIP process begins with a new idea for Ethereum Classic. Each potential ECIP must have '''a champion''' -- someone who writes the ECIP using the style and format described below, shepherds the discussions in the appropriate forums, and attempts to build community consensus around the idea. The '''ECIP champion (a.k.a. Author)''' should first attempt to ascertain whether the idea is ECIP-able.
+
+Small enhancements or patches to a particular piece of software often don't require standardization between multiple projects; these don't need an ECIP and should be injected into the relevant project-specific development workflow with a patch submission to the applicable issue tracker.
+
 Additionally, many ideas have been brought forward for changing Ethereum Classic that have been rejected for various reasons.
-The first step should be to search past discussions to see if an idea has been considered before, and if so, what issues arose in its progression.
-After investigating past work, the best way to proceed is by posting about the new idea to the [https://ethereumclassic.slack.com Ethereum Classic Slack Channel].
 
-Vetting an idea publicly before going as far as writing a ECIP is meant to save both the potential author and the wider community time.
+===Steps===
+
+'''1:''' The first step should be to search past discussions to see if an idea has been considered before, and if so, what issues arose in its progression.
+
+After investigating past work, the best way to proceed is by posting about the new idea to the [https://discord.gg/8q8xRtQ Ethereum Classic Discord #ecips Channel].
+
+Vetting an idea publicly before going as far as writing a ECIP is meant to save both the potential author and the '''wider community''' time.
+
 Asking the Ethereum Classic community first if an idea is original helps prevent too much time being spent on something that is guaranteed to be rejected based on prior discussions (searching the internet does not always do the trick).
+
 It also helps to make sure the idea is applicable to the entire community and not just the author. Just because an idea sounds good to the author does not mean it will work for most people in most areas where Ethereum Classic is used.
 
-Once the champion has asked the Ethereum Classic community as to whether an idea has any chance of acceptance, a draft ECIP should be submitted to the [https://github.com/ethereumproject/ECIPs ECIPs git repository] as a pull request. This gives the author a chance to flesh out the draft ECIP to make it properly formatted, of high quality, and to address additional concerns about the proposal. This draft must be written in ECIP style as described below, and named with an alias such as "ecip-johndoe-infinitebitcoins" until the editor has assigned it a ECIP number (authors MUST NOT self-assign ECIP numbers).
+'''2:''' Once the champion has asked the Ethereum Classic community as to whether an idea has any chance of acceptance, a '''draft ECIP''' should be submitted to the [https://github.com/ethereumproject/ECIPs ECIPs git repository] as a pull request. This gives the author a chance to flesh out the draft ECIP to make it properly formatted, of high quality, and to address additional concerns about the proposal. This draft must be written in ECIP style as described below, and named with an alias such as "ecip-johndoe-infinitecoins" until an editor has assigned it an ECIP number (authors MUST NOT self-assign ECIP numbers).
+
+ECIP authors '''are responsible for collecting community feedback''' on both the initial idea and the ECIP before submitting it for review. However, wherever possible, long open-ended discussions on public groups or mailing lists should be avoided. Strategies to keep the discussions efficient include: setting up a separate SIG mailing list for the topic, having the ECIP author accept private comments in the early design phases, setting up a wiki page or git repository, etc. ECIP authors should use their discretion here.
+
+It is highly recommended that '''a single ECIP contain a single key proposal or new idea'''. The more focused the ECIP, the more successful it tends to be. If in doubt, split your ECIP into several well-focused ones.
 
-ECIP authors are responsible for collecting community feedback on both the initial idea and the ECIP before submitting it for review. However, wherever possible, long open-ended discussions on public mailing lists should be avoided. Strategies to keep the discussions efficient include: setting up a separate SIG mailing list for the topic, having the ECIP author accept private comments in the early design phases, setting up a wiki page or git repository, etc. ECIP authors should use their discretion here.
+'''3:''' When the ECIP draft is complete, an ECIP editor will assign the ECIP a number, label it as Standards Track, Informational, or Process, and merge the pull request to the ECIPs git repository.
 
-It is highly recommended that a single ECIP contain a single key proposal or new idea. The more focused the ECIP, the more successful it tends to be. If in doubt, split your ECIP into several well-focused ones.
+The ECIP editor will not unreasonably reject an ECIP.
 
-When the ECIP draft is complete, the ECIP editor will assign the ECIP a number, label it as Standards Track, Informational, or Process, and merge the pull request to the ECIPs git repository.
-The ECIP editor will not unreasonably reject a ECIP.
-Reasons for rejecting BECPs include duplication of effort, disregard for formatting rules, being too unfocused or too broad, being technically unsound, not providing proper motivation or addressing backwards compatibility, or not in keeping with the Ethereum Classic philosophy.
-For a ECIP to be accepted it must meet certain minimum criteria.
-It must be a clear and complete description of the proposed enhancement.
-The enhancement must represent a net improvement.
-The proposed implementation, if applicable, must be solid and must not complicate the protocol unduly.
+Reasons for rejecting ECIPs include duplication of effort, disregard for formatting rules, being too unfocused or too broad, being technically unsound, not providing proper motivation or addressing backwards compatibility, or not in keeping with the Ethereum Classic philosophy.
 
-The ECIP author may update the draft as necessary in the git repository. Updates to drafts should also be submitted by the author as pull requests.
+For a ECIP to be accepted it must meet certain minimum criteria. It must be a clear and complete description of the proposed enhancement. The enhancement must represent a net improvement. The proposed implementation, if applicable, must be solid and must not complicate the protocol unduly.
+
+'''4:''' During the life cycle of the ECIP, the author may update the draft as necessary in the git repository. '''Updates to drafts should also be submitted by the author as pull requests.'''
 
 ===Transferring ECIP Ownership===
 
-It occasionally becomes necessary to transfer ownership of ECIPs to a new champion. In general, we'd like to retain the original author as a co-author of the transferred ECIP, but that's really up to the original author. A good reason to transfer ownership is because the original author no longer has the time or interest in updating it or following through with the ECIP process, or has fallen off the face of the 'net (i.e. is unreachable or not responding to email). A bad reason to transfer ownership is because you don't agree with the direction of the ECIP. We try to build consensus around a ECIP, but if that's not possible, you can always submit a competing ECIP.
+It occasionally becomes necessary to transfer ownership of ECIPs to a '''new champion'''. In general, we'd like to retain the original author '''as a co-author''' of the transferred ECIP, but that's really up to the original author. 
+
+A good reason to transfer ownership is because the original author no longer has the time or interest in updating it or following through with the ECIP process, or has fallen off the face of the net (i.e. is unreachable or not responding to email). A bad reason to transfer ownership is because you don't agree with the direction of the ECIP. We try to build consensus around an ECIP, but if that's not possible, you can always submit a competing ECIP.
 
-If you are interested in assuming ownership of a ECIP, send a message asking to take over, addressed to both the original author and the ECIP editor. If the original author doesn't respond to email in a timely manner, the ECIP editor will make a unilateral decision (it's not like such decisions can't be reversed :).
+'''If you are interested in assuming ownership of a ECIP''', send a message asking to take over, addressed to both the original author and the ECIP editor. If the original author doesn't respond to email in a timely manner, the ECIP editor will make a unilateral decision (it's not like such decisions can't be reversed :).
 
 ===ECIP Editors===
 
@@ -62,92 +74,95 @@ To be added.
 
 ===ECIP Editor Responsibilities & Workflow===
 
-For each new ECIP it should first be submitted as a "pull request" to the [https://github.com/ethereumproject/ECIPs ECIPs git repository] that comes in an editor does the following:
+====Introduction====
+
+ECIP editors are intended to fulfill administrative and editorial responsibilities. ECIP editors monitor ECIP changes, and update ECIP headers as appropriate.
 
-* Read the ECIP to check if it is ready: sound and complete. The ideas must make technical sense, even if they don't seem likely to be accepted.
+====Steps====
+
+'''1:''' Each new ECIP should first be submitted as a '''"pull request"''' to the [https://github.com/ethereumproject/ECIPs ECIPs git repository]. Then, an editor does the following:
+
+* Read the ECIP to check if it is ready: sound and complete. The ideas must make technical sense, even if it doesn't seem likely to be accepted.
 * The title should accurately describe the content.
 * Motivation and backward compatibility (when applicable) must be addressed.
 * The defined Layer header must be correctly assigned for the given specification.
 * Licensing terms must be acceptable for ECIPs.
 
-If the ECIP isn't ready, the editor will send it back to the author for revision, with specific instructions.
+'''2:''' If the ECIP isn't ready, the editor will send it back to the author for revision with specific instructions.
 
-Once the ECIP is ready, the ECIP editor will:
+'''3:''' Once the ECIP is ready, the ECIP editor will:
 
 * Assign a ECIP number in the pull request.
 * Merge the pull request when it is ready.
 * List the ECIP in [[README.mediawiki]]
 
-The ECIP editors are intended to fulfill administrative and editorial responsibilities. The ECIP editors monitor ECIP changes, and update ECIP headers as appropriate.
-
 ==ECIP format and structure==
 
 ===Specification===
 
-ECIPs should be written in mediawiki or markdown format.
+ECIPs should be written in '''mediawiki''' or '''markdown''' format.
 
 Each ECIP should have the following parts:
 
-* Preamble -- Headers containing metadata about the ECIP ([[#ECIP header preamble|see below]]).
-* Abstract -- A short (~200 word) description of the technical issue being addressed.
-* Copyright -- The ECIP must be explicitly licensed under acceptable copyright terms ([[#ECIP licensing|see below]]).
-* Specification -- The technical specification should describe the syntax and semantics of any new feature. The specification should be detailed enough to allow competing, interoperable implementations for any of the current Ethereum Classic platforms.
-* Motivation -- The motivation is critical for ECIPs that want to change the Ethereum Classic protocol. It should clearly explain why the existing protocol is inadequate to address the problem that the ECIP solves.
-* Rationale -- The rationale fleshes out the specification by describing what motivated the design and why particular design decisions were made. It should describe alternate designs that were considered and related work. The rationale should provide evidence of consensus within the community and discuss important objections or concerns raised during discussion.
-* Backwards compatibility -- All ECIPs that introduce backwards incompatibilities must include a section describing these incompatibilities and their severity. The ECIP must explain how the author proposes to deal with these incompatibilities.
-* Reference implementation -- The reference implementation must be completed before any ECIP is given status "Final", but it need not be completed before the ECIP is accepted. It is better to finish the specification and rationale first and reach consensus on it before writing code. The final implementation must include test code and documentation appropriate for the Ethereum Classic protocol.
+* '''Preamble''' -- Headers containing metadata about the ECIP [https://github.com/ethereumclassic/ECIPs/blob/master/ECIPs/ECIP-1000.mediawiki#ecip-header-preamble (see below)].
+* '''Abstract''' -- A short (~200 word) description of the technical issue being addressed.
+* '''Copyright''' -- The ECIP must be explicitly licensed under acceptable copyright terms [https://github.com/ethereumclassic/ECIPs/blob/master/ECIPs/ECIP-1000.mediawiki#ecip-licensing (see below)].
+* '''Specification''' -- The technical specification should describe the syntax and semantics of any new feature. It should be detailed enough to allow competing, interoperable implementations for any of the current Ethereum Classic platforms.
+* '''Motivation''' -- The motivation is critical for ECIPs that want to change the Ethereum Classic protocol. It should clearly explain why the existing protocol is inadequate to address the problem that the ECIP solves.
+* '''Rationale''' -- The rationale fleshes out the specification by describing what motivated the design and why particular design decisions were made. It should describe alternate designs that were considered and related work. It should provide evidence of consensus within the community and discuss important objections or concerns raised during discussion.
+* '''Backwards compatibility''' -- All ECIPs that introduce backwards incompatibilities must include a section describing these incompatibilities and their severity. The ECIP must explain how the author proposes to deal with these incompatibilities.
+* '''Reference implementation''' -- The reference implementation must be completed before any ECIP is given '''"Final"''' status, but it need not be completed before the ECIP is '''"Accepted"'''. It is better to finish the specification and rationale first and reach consensus on it before writing code. The final implementation must include test code and documentation appropriate for the Ethereum Classic protocol.
 
 ====ECIP header preamble====
 
 Each ECIP must begin with an RFC 822 style header preamble. The headers must appear in the following order. Headers marked with "*" are optional and are described below. All other headers are required.
 
-    ECIP: (ECIP number, or "?" before being assigned)
+  * ECIP: (ECIP number, or "?" before being assigned)
   * Layer: (Consensus (soft fork) | Consensus (hard fork) | Peer Services | API/RPC | Applications)
-    Title: (ECIP title; maximum 44 characters) 
-    Author: (authors real names and email addrs)
+  * Title: (ECIP title; maximum 44 characters) 
+  * Author: (authors real names and email addrs)
   * Discussions-To: (email address)
-    Comments-Summary: (summary tone)
-    Comments-URI: (links to wiki page for comments)
-    Status: (Draft | Active | Proposed | Deferred | Rejected |
-             Withdrawn | Final | Replaced | Obsolete)
-    Type: (Standards Track | Informational | Process)
-    Created: (date created on, in ISO 8601 (yyyy-mm-dd) format)
-    License: (abbreviation for approved license(s))
+  * Comments-Summary: (summary tone)
+  * Comments-URI: (links to wiki page for comments)
+  * Status: (Draft | Last Call | Accepted | Final | Deferred | Replaced | Rejected | Withdrawn)
+  * Type: (Standards Track | Informational | Process)
+  * Created: (date created on, in ISO 8601 (yyyy-mm-dd) format)
+  * License: (abbreviation for approved license(s))
   * License-Code: (abbreviation for code under different approved license(s))
   * Requires: (ECIP number(s))
   * Replaces: (ECIP number)
   * Superseded-By: (ECIP number)
 
-The Layer header (only for Standards Track ECIPs) documents which layer of Ethereum Classic the ECIP applies to.
+The ''Layer'' header (only for Standards Track ECIPs) documents which layer of Ethereum Classic the ECIP applies to.
 
-The Author header lists the names and email addresses of all the authors/owners of the ECIP.
-The format of the Author header value must be
+The ''Author'' header lists the names and email addresses of all the authors/owners of the ECIP.
+The format of the Author header value must be:
 
-  Random J. User <123@dom.ain>
+	Random J. User <123@dom.ain>
 
 If there are multiple authors, each should be on a separate line following RFC 2822 continuation line conventions.
 
-While an ECIP is in private discussions (usually during the initial Draft phase), a Discussions-To header will indicate the mailing list or URL where the ECIP is being discussed. No Discussions-To header is necessary if the ECIP is being discussed privately with the author.
+While an ECIP is in private discussions (usually during the initial Draft phase), a ''Discussions-To'' header will indicate the mailing list or URL where the ECIP is being discussed. No ''Discussions-To'' header is necessary if the ECIP is being discussed privately with the author.
 
-The Type header specifies the type of ECIP: Standards Track, Informational, or Process.
+The ''Type'' header specifies the type of ECIP: Standards Track, Informational, or Process.
 
-The Created header records the date that the ECIP was assigned a number.
+The ''Created'' header records the date that the ECIP was assigned a number.
 
-ECIPs may have a Requires header, indicating the ECIP numbers that this ECIP depends on.
+ECIPs may have a ''Requires'' header, indicating the ECIP numbers that this ECIP depends on.
 
-ECIPs may also have a Superseded-By header indicating that a ECIP has been rendered obsolete by a later document; the value is the number of the ECIP that replaces the current document. The newer ECIP must have a Replaces header containing the number of the ECIP that it rendered obsolete.
+ECIPs may also have a ''Superseded-By'' header indicating that a ECIP has been rendered obsolete by a later document; the value is the number of the ECIP that replaces the current document. The newer ECIP must have a ''Replaces'' header containing the number of the ECIP that it rendered obsolete.
 
 ====Auxiliary Files====
 
-ECIPs may include auxiliary files such as diagrams. Auxiliary files should be included in a subdirectory for that ECIP, or must be named ECIP-XXXX-Y.ext, where "XXXX" is the ECIP number, "Y" is a serial number (starting at 1), and "ext" is replaced by the actual file extension (e.g. "png").
+ECIPs may include auxiliary files such as diagrams. If if requires images, the image files should be included in a subdirectory of the ''assets'' folder for that ECIP as follow: ''assets/ecip-X''. When linking to an image in the ECIP, use the related links such as ''./assets/ecip-X/image.png''.
 
 ==ECIP types==
 
 There are three kinds of ECIP:
 
-* A Standards Track ECIP describes any change that affects most or all Ethereum Classic implementations, such as a change to the network protocol, a change in block or transaction validity rules, or any change or addition that affects the interoperability of applications using Ethereum Classic. Standards Track ECIPs consist of two parts, a design document and a reference implementation.
-* An Informational ECIP describes a Ethereum Classic design issue, or provides general guidelines or information to the Ethereum Classic community, but does not propose a new feature. Informational ECIPs do not necessarily represent a Ethereum Classic community consensus or recommendation, so users and implementors are free to ignore Informational ECIPs or follow their advice.
-* A Process ECIP describes a process surrounding Ethereum Classic, or proposes a change to (or an event in) a process. Process ECIPs are like Standards Track ECIPs but apply to areas other than the Ethereum Classic protocol itself. They may propose an implementation, but not to Ethereum Classic's codebase; they often require community consensus; unlike Informational ECIPs, they are more than recommendations, and users are typically not free to ignore them. Examples include procedures, guidelines, changes to the decision-making process, and changes to the tools or environment used in Ethereum Classic development. Any meta-ECIP is also considered a Process ECIP.
+* '''A Standards Track ECIP''' describes any change that affects most or all Ethereum Classic implementations, such as a change to the network protocol, a change in block or transaction validity rules, or any change or addition that affects the interoperability of applications using Ethereum Classic. Standards Track ECIPs consist of two parts, a design document and a reference implementation.
+* '''An Informational ECIP''' describes a Ethereum Classic design issue, or provides general guidelines or information to the Ethereum Classic community, but does not propose a new feature. Informational ECIPs do not necessarily represent a Ethereum Classic community consensus or recommendation, so users and implementors are free to ignore Informational ECIPs or follow their advice.
+* '''A Process ECIP''' describes a process surrounding Ethereum Classic, or proposes a change to (or an event in) a process. Process ECIPs are like Standards Track ECIPs but apply to areas other than the Ethereum Classic protocol itself. They may propose an implementation, but not to Ethereum Classic's codebase; they often require community consensus; unlike Informational ECIPs, they are more than recommendations, and users are typically not free to ignore them. Examples include procedures, guidelines, changes to the decision-making process, and changes to the tools or environment used in Ethereum Classic development. Any meta-ECIP is also considered a Process ECIP.
 
 ==ECIP status field==
 
@@ -155,20 +170,21 @@ There are three kinds of ECIP:
 
 The typical paths of the status of ECIPs are as follows:
 
-<img src="ECIP-1000/process.png"></img>
+  ./assets/ECIP-1000/process.png
+
+Champions of an ECIP may decide on their own to change the status between Draft, Deferred, or Withdrawn.
 
-Champions of a ECIP may decide on their own to change the status between Draft, Deferred, or Withdrawn.
 The ECIP editor may also change the status to Deferred when no progress is being made on the ECIP.
 
-A ECIP may only change status from Draft (or Rejected) to Proposed, when the author deems it is complete, has a working implementation (where applicable), and has community plans to progress it to the Final status.
+An ECIP may only change status from ''Draft'' (or ''Rejected'') to ''Last Call'', when the author deems it is complete, has a working implementation (where applicable), and has community plans to progress it to ''Final'' status.
 
-ECIPs should be changed from Draft or Proposed status, to Rejected status, upon request by any person, if they have not made progress in three years. Such a ECIP may be changed to Draft status if the champion provides revisions that meaningfully address public criticism of the proposal, or to Proposed status if it meets the criteria required as described in the previous paragraph.
+ECIPs should be changed from ''Draft'' or ''Last Call'' status, to ''Rejected'', upon request by any person, if they have not made progress in three years. Such a ECIP may be changed to ''Draft'' status if the champion provides revisions that meaningfully address public criticism of the proposal, or to ''Last Call'' if it meets the criteria required as described in the previous paragraph.
 
-An Proposed ECIP may progress to Final only when specific criteria reflecting real-world adoption has occurred. This is different for each ECIP depending on the nature of its proposed changes, which will be expanded on below. Evaluation of this status change should be objectively verifiable, and/or be discussed on the development mailing list.
+An ''Accepted'' ECIP may progress to ''Final'' only when specific criteria reflecting real-world adoption has occurred. This is different for each ECIP depending on the nature of its proposed changes, which will be expanded on below. Evaluation of this status change should be '''objectively verifiable''', and/or be discussed on the development calls, Discord channel, other groups or the mailing list.
 
-When a Final ECIP is no longer relevant, its status may be changed to Replaced or Obsolete (which is equivalent to Replaced). This change must also be objectively verifiable and/or discussed.
+When a ''Final'' ECIP is no longer relevant, its status may be changed to ''Replaced''. This change must also be '''objectively verifiable''' and/or discussed.
 
-A process ECIP may change status from Draft to Active when it achieves rough consensus on the mailing list. Such a proposal is said to have rough consensus if it has been open to discussion on the development mailing list for at least one month, and no person maintains any unaddressed substantiated objections to it. Addressed or obstructive objections may be ignored/overruled by general agreement that they have been sufficiently addressed, but clear reasoning must be given in such circumstances.
+A process ECIP may change status from ''Draft'' to ''Final'' when it achieves rough consensus on the discussion process. Such a proposal is said to have rough consensus if it has been open to discussion on the development calls, Discord channel, other groups or the mailing list for at least one month, and no person maintains any unaddressed substantiated objections to it. Addressed or obstructive objections may be ignored/overruled by general agreement that they have been sufficiently addressed, but clear reasoning must be given in such circumstances.
 
 ====Progression to Final status====
 
@@ -176,26 +192,26 @@ Peer services ECIPs should be observed to be adopted by at least 1% of public li
 
 API/RPC and application layer ECIPs must be implemented by at least two independent and compatible software applications.
 
-Software authors are encouraged to publish summaries of what ECIPs their software supports to aid in verification of status changes. Good examples of this at the time of writing this ECIP, can be observed in [https://github.com/bitcoin/bitcoin/blob/master/doc/BIPs.md Bitcoin Core's doc/BIPs.md file] as well as [https://github.com/schildbach/bitcoin-wallet/blob/master/wallet/README.specs Bitcoin Wallet for Android's wallet/README.specs file].
+Software authors are encouraged to publish summaries of what ECIPs their software supports to aid in verification of status changes. Good examples of this, at the time of writing this ECIP, can be observed in [https://github.com/bitcoin/bitcoin/blob/master/doc/BIPs.md Bitcoin Core's doc/BIPs.md file] as well as [https://github.com/schildbach/bitcoin-wallet/blob/master/wallet/README.specs Bitcoin Wallet for Android's wallet/README.specs file].
 
-These criteria are considered objective ways to observe the de facto adoption of the ECIP, and are not to be used as reasons to oppose or reject a ECIP. Should a ECIP become actually and unambiguously adopted despite not meeting the criteria outlined here, it should still be updated to Final status.
+These criteria are considered objective ways to observe the de facto adoption of the ECIP, and are not to be used as reasons to oppose or reject an ECIP. Should an ECIP become actually and unambiguously adopted despite not meeting the criteria outlined here, it should still be updated to `Final` status.
 
 ===Rationale===
 
-Why is this necessary at all?
+'''Why is this necessary at all?'''
 
-* Many ECIPs with significant real-world use have been left as Draft or Proposed status longer than appropriate. By giving objective criteria to judge the progression of ECIPs, this proposal aims to help keep the Status accurate and up-to-date.
+* Many ECIPs with significant real-world use have been left as ''Draft'' or ''Last Call'' status longer than appropriate. By giving objective criteria to judge the progression of ECIPs, this proposal aims to help keep the '''Status''' accurate and up-to-date.
 
-What is the ideal percentage of listening nodes needed to adopt peer services proposals?
+'''What is the ideal percentage of listening nodes needed to adopt peer services proposals?'''
 
 * This is unknown, and set rather arbitrarily at this time. For a random selection of peers to have at least one other peer implementing the extension, 13% or more would be necessary, but nodes could continue to scan the network for such peers with perhaps some reasonable success. Furthermore, service bits exist to help identification upfront.
 
-Why is it necessary for at least two software projects to release an implementation of API/RPC and application layer ECIPs, before they become Final?
+'''Why is it necessary for at least two software projects to release an implementation of API/RPC and application layer ECIPs, before they become Final?'''
 
 * If there is only one implementation of a specification, there is no other program for which a standard interface is used with or needed.
 * Even if there are only two projects rather than more, some standard coordination between them exists.
 
-What if a ECIP is proposed that only makes sense for a single specific project?
+'''What if a ECIP is proposed that only makes sense for a single specific project?'''
 
 * The ECIP process exists for standardisation between independent projects. If something only affects one project, it should be done through that project's own internal processes, and never be proposed as a ECIP in the first place.
 
@@ -204,14 +220,20 @@ What if a ECIP is proposed that only makes sense for a single specific project?
 ===Specification===
 
 Each ECIP should, in its preamble, link to a public wiki page with a summary tone of the comments on that page.
+
 Reviewers of the ECIP who consider themselves qualified, should post their own comments on this wiki page.
+
 The comments page should generally only be used to post final comments for a completed ECIP.
-If a ECIP is not yet completed, reviewers should instead post on the applicable development mailing list thread to allow the ECIP author(s) to address any concerns or problems pointed out by the review.
+
+If an ECIP is not yet completed, reviewers should instead post on the applicable development Discord group or mailing list thread to allow the ECIP author(s) to address any concerns or problems pointed out by the review.
 
 Some ECIPs receive exposure outside the development community prior to completion, and other ECIPs might not be completed at all. To avoid a situation where critical ECIP reviews may go unnoticed during this period, reviewers may, at their option, still post their review on the comments page, provided they first post it to the mailing list and plan to later remove or revise it as applicable based on the completed version. Such revisions should be made by editing the previous review and updating the timestamp. Reviews made prior to the complete version may be removed if they are no longer applicable and have not been updated in a timely manner (eg, within one month).
 
 Pages must be named after the full ECIP number (eg, "ECIP 0001") and placed in the "Comments" namespace.
-For example, the link for ECIP 1 will be https://github.com/ethereumproject/ECIPs/wiki/Comments:ECIP-0001 .
+
+For example, the link for ECIP 1 will be:
+
+    https://github.com/ethereumproject/ECIPs/wiki/Comments:ECIP-0001
 
 Comments posted to this wiki should use the following format:
 
@@ -235,19 +257,19 @@ For example, the preamble to ECIP 1 might be updated to include the line:
     Comments-URI: https://github.com/ethereumproject/ECIPs/wiki/Comments:ECIP-1000
                   https://some-other-wiki.org/ECIP_1_Comments
 
-These fields must follow the "Discussions-To" header defined in ECIP 1 (if that header is not present, it should follow the position where it would be present; generally this is immediately above the Status header).
+These fields must follow the **Discussions-To** header defined in ECIP 1 (if that header is not present, it should follow the position where it would be present; generally this is immediately above the **Status** header).
 
-To avoid doubt: comments and status are unrelated metrics to judge a ECIP, and neither should be directly influencing the other.
+To avoid doubt: comments and status are unrelated metrics to judge an ECIP, and neither should be directly influencing the other.
 
 ===Rationale===
 
-What is the purpose of ECIP comments?
+'''What is the purpose of ECIP comments?'''
 
-* Various ECIPs have been adopted (the criteria required for "Final" Status) despite being considered generally inadvisable. Some presently regard ECIPs as a "good idea" simply by virtue of them being assigned a ECIP number. Due to the low barrier of entry for submission of new ECIPs, it seems advisable for a way for reviewers to express their opinions on them in a way that is consumable to the public without needing to review the entire development discussion.
+- Various ECIPs have been adopted (the criteria required for ''Final'' status) despite being considered generally inadvisable. Some presently regard ECIPs as a "good idea" simply by virtue of them being assigned an ECIP number. Due to the low barrier of entry for submission of new ECIPs, it seems advisable for a way for reviewers to express their opinions on them in a way that is consumable to the public without needing to review the entire development discussion.
 
-Will ECIP comments be censored or limited to particular participants/"experts"?
+'''Will ECIP comments be censored or limited to particular participants/"experts"?'''
 
-* Participants should freely refrain from commenting outside of their area of knowledge or expertise. However, comments should not be censored, and participation should be open to the public.
+- Participants should freely refrain from commenting outside of their area of knowledge or expertise. However, comments should not be censored, and participation should be open to the public.
 
 ==ECIP licensing==
 
@@ -271,7 +293,9 @@ For example, a preamble specifying the optional License-Code header might look l
     License-Code: GPL-2.0+
 
 In this case, the code in the ECIP is not available under the BSD or All-Permissive licenses, but only under the terms of the GNU General Public License (GPL), version 2 or newer.
+
 If the code were to be available under *only* version 2 exactly, the "+" symbol should be removed from the license abbreviation.
+
 For a later version (eg, GPL 3.0), you would increase the version number (and retain or remove the "+" depending on intent).
 
     License-Code: GPL-2.0   # This refers to GPL v2.0 *only*, no later license versions are acceptable.
@@ -282,6 +306,7 @@ For a later version (eg, GPL 3.0), you would increase the version number (and re
 In the event that the licensing for the text or code is too complicated to express with a simple list of alternatives, the list should instead be replaced with the single term "Complex". In all cases, details of the licensing terms must be provided in the Copyright section of the ECIP.
 
 ECIPs are not required to be *exclusively* licensed under approved terms, and may also be licensed under unacceptable licenses *in addition to* at least one acceptable license.
+
 In this case, only the acceptable license(s) should be listed in the License and License-Code headers.
 
 ====Recommended licenses====
@@ -307,7 +332,7 @@ In addition, it is recommended that literal code included in the ECIP be dual-li
 
 ===Rationale===
 
-Why are there software licenses included?
+'''Why are there software licenses included?'''
 
 * Some ECIPs, especially consensus layer, may include literal code in the ECIP itself which may not be available under the exact license terms of the ECIP.
 * Despite this, not all software licenses would be acceptable for content included in ECIPs.

From 10b7010d2ad9cb8ca9cf5b54e5e42b1bfe226325 Mon Sep 17 00:00:00 2001
From: Donald McIntyre <newfination@gmail.com>
Date: Thu, 27 Jun 2019 20:30:39 -0600
Subject: [PATCH 2/2] Corrected ECIP number

Corrected ECIP number from 2 to 1000 in the preamble.
---
 ECIPs/ECIP-1000.mediawiki | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/ECIPs/ECIP-1000.mediawiki b/ECIPs/ECIP-1000.mediawiki
index d0bac6f9..5a9fdb15 100644
--- a/ECIPs/ECIP-1000.mediawiki
+++ b/ECIPs/ECIP-1000.mediawiki
@@ -1,5 +1,5 @@
 <pre>
-  ECIP: 2
+  ECIP: 1000
   Title: ECIP process
   Author: Wei Tang <hi@that.world>
   Comments-Summary: No comments yet.
